### PR TITLE
Run address sanitiser in the GitHub CI

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

**Comment by markshannon** [comment]
Looks like you've found an X11 bug.
I don't where the X11 bug tracker is. Somewhere in https://gitlab.freedesktop.org/ probably.

**Comment by markshannon** [comment]
I'd exclude the failing tests for now and add a note somewhere to revisit the exclusions in a month or two.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -218,3 +218,46 @@ jobs:
...
+    - name: Configure ccache action
+      uses: hendrikmuhs/ccache-action@v1
+    - name: Configure CPython
+      run: ./configure --with-address-sanitizer --without-pymalloc
+    - name: Build CPython
+      run: make -j4
+    - name: Display build info
+      run: make pythoninfo
+    - name: Tests
+      run: xvfb-run make buildbottest TESTOPTS="-j4 -uall,-cpu -x test_ctypes test_capi test_crypt test_decimal test_faulthandler test_interpreters"
```
Comment (markshannon) [inline .github/workflows/build.yml]: Add `test___all__ test_idle test_tix test_tk test_ttk_guionly test_ttk_textonly` to this list.

Comment (pablogsal) [inline .github/workflows/build.yml]:
```suggestion
      run: xvfb-run make buildbottest TESTOPTS="-j4 -uall,-cpu -x test_ctypes test_crypt test_decimal test_faulthandler test_interpreters test___all__ test_idle test_tix test_tk test_ttk_guionly test_ttk_textonly"
```

**Comment by vstinner** [comment]
Rather than putting ``-x test_ctypes test_crypt test_decimal test_faulthandler test_interpreters test___all__ test_idle test_tix test_tk test_ttk_guionly test_ttk_textonly`` list in the CI configuration, what do you think of adding an option to regrtest, similar to --pgo? Example: add ``--asan`` option.

**Comment by pablogsal** [comment]
> Rather than putting `-x test_ctypes test_crypt test_decimal test_faulthandler test_interpreters test___all__ test_idle test_tix test_tk test_ttk_guionly test_ttk_textonly` list in the CI configuration, what do you think of adding an option to regrtest, similar to --pgo? Example: add `--asan` option.

👍  Makes sense, that's clearly a better option

**Comment by markshannon** [comment]
`--pgo` describes what this does, whereas `--asan` would not.
I know it is clumsy to have to list all the exceptions, but that makes it clear what is excluded.
We want to make the list shorter, having this long list makes it clear what needs fixing.

**Comment by pablogsal** [comment]
Yeah, that makes sense, although it may be worth to also add *some* option so you can easily run this locally. I think we should also add something to the devguide about what is excluded and why.

In any case, I am going to merge this and add the option to regrtest in a separate PR.

**Comment by vstinner** [comment]
> I know it is clumsy to have to list all the exceptions, but that makes it clear what is excluded.

My concern is that there is also an ASAN buildbot, and this denylist is copied manually. It increases the maintenance burden. Well, since @pablogsal runs the buildbot, he is likely eagger to maintain these two lists manually ;-)

Maybe an option is not needed, and regrtest could detected if Python was built with ASAN using sysconfig to get compiler flags.

>  We want to make the list shorter, having this long list makes it clear what needs fixing.

Sure, it would be great to skip less tests ;-)

**Pull request merged**